agenta-backend/agenta_backend/routers/app_router.py

```diff
@@ -340,20 +341,33 @@ async def create_app_and_variant_from_template(
                 app_name, organization_id, **user_org_data
             )
 
+        logger.debug("Step 5 (extra): Retrieve template from db")
+        template_db = await db_manager.get_template(payload.template_id)
+
         logger.debug(
             "Step 6: Creating image instance and adding variant based on image"
         )
-        image_name = f"agentaai/templates:{payload.image_tag}"
+        image_name = f"agentaai/templates:{template_db.name}"
         app_variant_db = await app_manager.add_variant_based_on_image(
             app=app,
             variant_name="app.default",
-            docker_id=payload.image_id,
+            docker_id=template_db.digest,
             tags=f"{image_name}",
             base_name="app",
             config_name="default",
             **user_org_data,
         )
 
+        if os.environ["FEATURE_FLAG"] == "oss":
+            logger.debug("Step 6 (extra): Pulling image from Docker Hub")
+            repo_owner, repo_name = (
+                settings.docker_hub_repo_owner,
+                settings.docker_hub_repo_name,
+            )
+            await container_manager.pull_docker_image(
```

Comment:
@aybruhm why do we need this step? Did we not do it already at the start of the app?

Reply:
Ah, yes. True. That skipped my mind.